src/client.ts

```diff
@@ -119,15 +127,22 @@ export class Client {
    * Performs a HTTP request to the endpoint specified with the default
    * options of the client.
    *
-   * @param  {String}   resource Resource to be called by GET.
+   * @param  {String}   resource Resource to be called by GET and POST.
+   * @param  {object}   payload  Resource to be called by POST.
    * @return {Promise<Response>}
    */
-  public async request(resource: string): Promise<Response> {
-    const response = await fetch(resource, {
-      method: 'GET',
+  public async request(resource: string, payload?: object): Promise<Response> {
+    const method: string = typeof payload === 'undefined' ? 'GET' : 'POST';
+    const requestInit: GenericObject = {
+      method: method,
       mode: 'cors',
       headers: Object.assign({}, this.headers),
-    });
+    };
+    if (method == 'POST') {
+      requestInit['body'] = JSON.stringify(payload);
+    }
+    const init = Object.assign({}, requestInit);
```

Comment:
This is not required, you created `requestInit` so no need to protect it from rewriting :wink: (and Object.assign does not deep copy, so…).